drgn_tools/nfs_tools.py

```diff
@@ -0,0 +1,931 @@
+
+    :param prog: drgn program
+    :param super_block: struct super_block
+    :returns: struct mount associated with the specified super_block
+    """
+
+    prog = super_block.prog_
+    ns = prog["init_task"].nsproxy.mnt_ns
+    for mnt in list_for_each_entry(
+        "struct mount", ns.list.address_of_(), "mnt_list"
+    ):
+        if mnt.mnt.mnt_sb == super_block:
+            return mnt
+    return None
+
+
+def get_ipaddr_str(sockaddr: Object) -> str:
+    """
+    Convert struct sockaddr_storage to IP address string.
+    Only support IPv4 and IPv6 address family.
+
+    :param addr: struct sockaddr_storage
+    :returns: IP address string
+    """
+
+    prog = sockaddr.prog_
+    if sockaddr.ss_family == 2:
+        """AF_INET IPv4"""
+        si = Object(
+            prog, "struct sockaddr_in *", value=sockaddr.address_of_().value_()
+        )
+        ip = str(
+            ipaddress.IPv4Address(socket.htonl(si.sin_addr.s_addr.value_()))
+        )
+        port = str(socket.htons(si.sin_port.value_()))
+        return ip + ":" + port
+    elif sockaddr.ss_family == 10:
+        si = Object(
+            prog, "struct in6_addr *", value=sockaddr.address_of_().value_()
+        )
+        ip = rds_inet_ntoa(si)
+        """ AF_INET6 IPv6 """
+        return ip
+    elif sockaddr.ss_family == 1:
+        """AF_UNIX Unix domain socket"""
+        return "Unix domain socket"
+    else:
+        return (
+            "Address Family["
+            + str(sockaddr.ss_family.value_())
+            + "] NOT_SUPPORTED"
+        )
+
+
+def count_nodes(list_head: Object) -> int:
+    """
+    Return the number of entries in the given list_head.
+
+    :param list_head: struct list_head.
+    :returns: node count
+    """
+
+    n = 0
+    head = list_head.address_of_()
+    node = head.next
+    while node != head:
+        n += 1
+        node = node.next
+    return n
+
+
+def lookup_ksym(prog: drgn.Program, addr: int) -> str:
+    """
+    Look up the symbol associated with the given kernel address.
+
+    :param prog: drgn program
+    :param addr: kernel address
+    :returns: the symbol associated with the kernel address.
+             If the symbol is not available, return the address
+             as a string in hex.
+    """
+
+    if addr == 0:
+        return "NULL"
+    try:
+        proc_sym = prog.symbol(addr).name
+    except LookupError:
+        proc_sym = str(hex(addr))
+    return proc_sym
+
+
+"""
+---------------------- General helpers ------------------------------
+These helpers will be deleted if there are already similiar functions
+in drgn_tools or drgn.helper
+---------------------------------------------------------------------
+"""
+
+
+def panic_task(prog: drgn.Program) -> Object:
+    """
+    Return the task_struct that causes the panic.
+
+    :param prog: drgn program
+    :returns: task_struct object
+    """
+
+    return per_cpu(prog["runqueues"], prog["crashing_cpu"]).curr
```

Comment:
You can use `prog.crashed_thread()` to get the `drgn.Thread` which is crashing. `Thread.object` will be a `struct task_struct *`.

Even aside from that, it's important to note that `prog["crashing_cpu"]` is x86_64 specific, and we support aarch64, so if you were implementing this, I would recommend `prog["panic_cpu"]`.